src/v/cluster/archival_metadata_stm.h

```diff
@@ -99,7 +99,7 @@ class archival_metadata_stm final : public persisted_stm {
 private:
     prefix_logger _logger;
 
-    mutex _lock;
+    mutex _lock{"c/archive/metadata-stm"};
```

Comment:
I created another issue for this work--was planning on doing it after this gets in. I prefer small, short-lived PRs.